cmd/protoc-gen-go/internal_gengo/main.go

```diff
@@ -843,6 +843,44 @@ func genMessageOneofWrapperTypes(g *protogen.GeneratedFile, f *fileInfo, m *mess
 				trailingComment(field.Comments.Trailing))
 			g.P("}")
 			g.P()
+
+			ty, _ := fieldGoType(g, f, field)
+			ty = strings.ReplaceAll(ty, "*", "")
+			var args []string
+			if field.Message != nil {
+				// oneof 가 메시지인 경우 해당 메시지의 1-depth 필드로 생성자함수를 만들어준다
+				args = make([]string, len(field.Message.Fields))
+				for i := 0; i < len(field.Message.Fields); i++ {
+					ty, isPtr := fieldGoType(g, f, field.Message.Fields[i])
+					if isPtr {
+						args[i] = fmt.Sprintf("p%s *%s", field.Message.Fields[i].GoName, ty)
+					} else {
+						args[i] = fmt.Sprintf("p%s %s", field.Message.Fields[i].GoName, ty)
+					}
+				}
+			} else {
+				// oneof 가 메시지가 아닌경우 해당 타입으로 생성자함수를 만들어준다
+				//args = []string{fmt.Sprintf("v %s", field.GoIdent.GoName)}
+				args = []string{fmt.Sprintf("p %s", ty)}
+			}
```

Comment:
```suggestion
			// oneof 가 메시지가 아닌경우 해당 타입으로 생성자함수를 만들어준다
			args = []string{fmt.Sprintf("p %s", ty)}
			if field.Message != nil {
				// oneof 가 메시지인 경우 해당 메시지의 1-depth 필드로 생성자함수를 만들어준다
				args = make([]string, len(field.Message.Fields))
				for i := 0; i < len(field.Message.Fields); i++ {
					args[i] = makeArguments(g, f, field.Message.Fields[i])
				}
			}
```

이런식으로 되면 분기가 덜 보일 것 같아요

Reply:
아니면 makeArguments로 해서 for문이 안보이게 해도 될것같구요

Reply:
[57e3fd8](https://github.com/devsisters/protobuf-go/pull/1/commits/57e3fd8b17f2094eb9cf1085fd69161475cdd476)

뭔가 사실 두 분기가 헤드,테일만 비슷하고 다른 코드인 것 같아서 분기를 아예 나누어보았습니다